ui/ducks/metamask/metamask.js

```diff
@@ -344,10 +346,36 @@ export function getGasEstimateType(state) {
   return state.metamask.gasEstimateType;
 }
 
-export function getGasFeeEstimates(state) {
+export function getGasFeeControllerEstimates(state) {
   return state.metamask.gasFeeEstimates;
 }
 
+export function getTransactionGasFeeEstimates(state) {
+  const transactionMetadata = state.confirmTransaction?.txData;
+  return transactionMetadata?.gasFeeEstimates;
+}
+
+export const getGasFeeEstimates = createSelector(
+  getGasEstimateType,
+  getGasFeeControllerEstimates,
+  getTransactionGasFeeEstimates,
+  (
+    gasFeeControllerEstimateType,
+    gasFeeControllerEstimates,
+    transactionGasFeeEstimates,
+  ) => {
+    if (transactionGasFeeEstimates) {
```

Comment:
Matt, would it make sense to include the check for non-null `transactionGasFeeEstimates` directly within `mergeGasFeeEstimates`? This could act as a useful precondition check. It would also simplify call sites, like this selector:

```typescript
export const getGasFeeEstimates = createSelector(
  getGasEstimateType,
  getGasFeeControllerEstimates,
  getTransactionGasFeeEstimates,
  mergeGasFeeEstimates,
);
```

Reply:
I did consider that, but I didn't want to overstep the responsibility of the controller.

It's awkward logic to merge the two types of estimates and handle different number formats and denominations so a utility function was warranted.

However it's ultimately a client decision which they want to use, so it felt safer to make that an explicit decision in the selectors.